addons/sale/migrations/9.0.1.0/post-migration.py

```diff
@@ -139,18 +142,19 @@ def set_so_line_computed_rest(env):
         env.cr, """
         UPDATE sale_order_line sol
         SET invoice_status = CASE
-            WHEN sol.state IN ('sale', 'done') THEN 'no'
-            WHEN abs(sol.qty_to_invoice) > 0.00001 THEN 'to invoice'
+            WHEN sol.state NOT IN ('sale', 'done') THEN 'no'
+            WHEN ROUND(sol.qty_to_invoice, %(uom_precision)s) != 0 THEN 'to invoice'
             WHEN sol.state = 'sale' AND pt.invoice_policy = 'order' AND
-                (sol.qty_delivered - sol.product_uom_qty) > 0.00001
+                ROUND(sol.qty_delivered, %(uom_precision)s) > ROUND(sol.product_uom_qty, %(uom_precision)s)
                     THEN 'upselling'
-            WHEN sol.qty_invoiced >= sol.product_uom_qty THEN 'invoiced'
+            WHEN ROUND(sol.qty_invoiced, %(uom_precision)s) = ROUND(sol.product_uom_qty, %(uom_precision)s) THEN 'invoiced'
             ELSE 'no'
         END
         FROM sale_order_line sol2
             JOIN product_product pp ON pp.id = sol2.product_id
             JOIN product_template pt ON pt.id = pp.product_tmpl_id
-        WHERE sol2.id = sol.id""")
+        WHERE sol2.id = sol.id""",
+        {"uom_precision": uom_precision})
```

Comment:
Should this be handled by psycopg2 argument substitution or directly by Python?

Reply:
The result is an integer, so it should be OK as it is AFAIK. See:

https://github.com/OCA/OpenUpgrade/blob/bd1a1d7b04b1d5ea5d94b931443d61fc71d810a2/addons/decimal_precision/decimal_precision.py#L10-L15

https://github.com/OCA/OpenUpgrade/blob/bd1a1d7b04b1d5ea5d94b931443d61fc71d810a2/addons/decimal_precision/decimal_precision.py#L24-L28